src/azul/openapi/validation.py

```diff
@@ -0,0 +1,218 @@
+    Mapping,
+    Optional,
+)
+
+import jsonschema
+from more_itertools import (
+    one,
+)
+
+from azul import (
+    RequirementError,
+    reject,
+)
+from azul.openapi.schema import (
+    type_lookup,
+)
+from azul.plugins.metadata.hca.transform import (
+    value_and_unit,
+)
+from azul.types import (
+    JSON,
+)
+from azul.uuids import (
+    InvalidUUIDError,
+    InvalidUUIDVersionError,
+    validate_uuid,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ValidationError(Exception):
+
+    def __init__(self,
+                 message,
+                 name,
+                 value,
+                 schema: Optional[JSON] = None):
+        self.message = message
+        self.name = name
+        self.value = value
+        if schema is not None:
+            self.schema = schema
+
+
+class SpecValidator:
+
+    def __init__(self,
+                 *,
+                 name: str,
+                 required: bool,
+                 default: str = None,
+                 schema: Mapping[str, str] = None,
+                 content: Mapping[str, str] = None,
+                 pattern: Optional[str] = None,
+                 validator: Callable,
+                 **kwargs):
+        """
+        :param kwargs: Since 'in' is a Python reserved word we can only extract it
+         using kwargs. All other kwargs are ignored.
+        """
+        self.in_ = kwargs['in']
+        self.json_validators = dict(jsonschema.Draft4Validator.VALIDATORS)
+        self.name = name
+        self.required = required
+        self.pattern = pattern
+        self.validator = validator
+        self.default = default
+        self.schema = schema
+        self.content = content
+
+    def validate(self, param_value: str):
+        raise NotImplementedError()
+
+
+class SchemaSpecValidator(SpecValidator):
+
+    def __init__(self,
+                 *,
+                 name: str,
+                 required: bool,
+                 pattern: Optional[str] = None,
+                 schema: Mapping,
+                 **kwargs):
+        reject('content' in kwargs)
+        super().__init__(name=name,
+                         required=required,
+                         default=schema.get('default'),
+                         schema=schema,
+                         pattern=pattern,
+                         validator=type_lookup(schema['type']),
+                         **kwargs)
+
+    def validate(self, param_value: str):
+        """
+        Validates if a given request parameter meets the specification.
+        :param param_value: The value of the parameter to be validated
+        """
+        try:
+            self.validator(param_value)
+        except (TypeError, ValueError):
+            raise ValidationError('Invalid parameter', self.name, param_value)
+        schema_type = self.schema['type']
+        if schema_type == 'string':
+            if self.schema.get('format') == 'uuid':
+                # noinspection PyUnusedLocal
```

Comment:
prepend underscore to name of unused variable that also gets rid of the warning.

Reply:
this section was removed in favor of pattern specification for UUIDs